code/model/Translatable.php

```diff
@@ -1212,7 +1212,7 @@ protected function addTranslatableFields(&$fields) {
 				if(in_array($dataField->getName(), $translatableFieldNames)) {
 					// if the field is translatable, perform transformation
 					$fields->replaceField($dataField->getName(), $transformation->transformFormField($dataField));
-				} elseif(!$dataField->isReadonly()) {
+				} elseif($dataField->isReadonly()) {
```

Comment:
The comment under this line basically says that if it's not readonly then make it read only, so this change basically skips this logic block